PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -50,33 +79,130 @@ public void onResult(Configuration configuration, Exception error) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.error");
+                        callback.onResult(null, new BraintreeException(errorInfo.getReason()));
+                    }
+                }
+        );
+    }
+
+    //TODO: I copied all of this from PayPal client and adapted it. Is all of this necessary?
+    private void approved(final Approval approval, final PayPalResponse payPalResponse, final PayPalNativeFlowCallback callback) {
+        Uri deepLinkUri = Uri.parse(String.format(
+                "%s://onetouch/v1/success?paymentId=%s&token=%s&PayerID=%s",
+                braintreeClient.getReturnUrlScheme(),
+                approval.getData().getPaymentId(),
+                approval.getData().getOrderId(),
+                approval.getData().getPayerId()));
+        try {
+            JSONObject urlResponseData = parseUrlResponseData(deepLinkUri, payPalResponse.getSuccessUrl(), payPalResponse.getApprovalUrl(), "token");
+            PayPalAccount payPalAccount = new PayPalAccount();
+            payPalAccount.setClientMetadataId(payPalResponse.getClientMetadataId());
+            payPalAccount.setSource("paypal-browser"); // paypal native throws error
+            payPalAccount.setUrlResponseData(urlResponseData);
+            payPalAccount.setPaymentType("single-payment"); // is this the correct way?
+
+            if (payPalResponse.getMerchantAccountId() != null) {
+                payPalAccount.setMerchantAccountId(payPalResponse.getMerchantAccountId());
+            }
+
+            if (payPalResponse.getIntent() != null) {
+                payPalAccount.setIntent(payPalResponse.getIntent());
+            }
+            tokenizationClient.tokenizeREST(payPalAccount, new TokenizeCallback() {
+                @Override
+                public void onResult(JSONObject tokenizationResponse, Exception exception) {
+                    if (tokenizationResponse != null) {
+                        try {
+                            PayPalAccountNonce payPalAccountNonce = PayPalAccountNonce.fromJSON(tokenizationResponse);
+                            if (payPalAccountNonce.getCreditFinancing() != null) {
+                                braintreeClient.sendAnalyticsEvent("paypal.credit.accepted");
+                            }
+                            callback.onResult(payPalAccountNonce, null);
+                        } catch (JSONException e) {
+                            callback.onResult(null, e);
+                        }
+                    } else {
+                        callback.onResult(null, exception);
+                    }
+                }
+            });
+        } catch (final JSONException | BraintreeException e) {
+            callback.onResult(null, e);
+        }
+
+    }
+
+    private String getPayPalReturnUrl(Context context) {
+        //Shouldnt the configuration include the return URL of the client? why are generating this?
```

Comment:
I don't think this return URL is currently included in the fetched configuration response, so I think we do need to generate this here for now.